extensions/db-diagram/src/models/Generics/INetworkGraphModel.ts

```diff
@@ -0,0 +1,16 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import INodeModel from './INodeModel';
+
+
+export default interface INetworkGraphModel {
+	name: string;
+	summary: string;
+	nodes: INodeModel[];
+	getName(): string;
```

Comment:
You can get rid of the `getXXX()` methods everywhere, since you can directly access them